---
title: "C26110 | Microsoft Docs"
ms.custom: ""
ms.date: "12/05/2016"
ms.prod: "visual-studio-dev14"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "vs-devops-test"
ms.tgt_pltfrm: ""
ms.topic: "article"
f1_keywords: 
  - "C26110"
helpviewer_keywords: 
  - "C26110"
ms.assetid: d82b79ec-6d7f-438b-bd6a-da874a3e08e5
caps.latest.revision: 8
caps.handback.revision: 8
author: "corob-msft"
ms.author: "corob"
manager: "ghogen"
---
# C26110
[!INCLUDE[vs2017banner](../code-quality/includes/vs2017banner.md)]

Advertencia C26110: Error de llamada al retener un bloqueo \<bloqueo\> antes de llamar a la función \<función\>.  
  
 Cuando se requiere un bloqueo, asegúrese de clarificar si la propia función o su llamador adquiere el bloqueo.  La advertencia C26110 se emite cuando hay una infracción de anotación de `_Requires_lock_held_`.  
  
## Ejemplo  
 En el ejemplo siguiente, se genera la advertencia C26110 porque la anotación `_Requires_lock_held_` en la función `LockRequired` establece que el llamador `LockRequired` debe adquirir el bloqueo antes de llamar a `LockRequired`.  Sin esta anotación, `LockRequired` debe adquirir el bloqueo antes de obtener acceso a datos compartidos protegidos por el bloqueo.  
  
```  
  
typedef struct _DATA   
{  
    CRITICAL_SECTION cs;  
    int d;  
} DATA;  
  
_Requires_lock_held_(p->cs)  
  
void LockRequired(DATA* p)  
{  
    p->d = 0;  
}  
  
void LockNotHeld(DATA* p)   
{   
    LockRequired(p); // Warning C26110   
}  
  
```